src/game_inventory.cpp

```diff
@@ -343,7 +341,7 @@ class comestible_inventory_preset : public inventory_selector_preset
 
     protected:
         int rate_freshness( const item &it, const item &container ) const {
-            if( p.will_eat( it ) == edible_rating::ROTTEN ) {
+            if( *p.will_eat( it ) == edible_rating::ROTTEN ) {
```

Comment:
Would be clearer without this overload thing IMO. This overload requires understanding the type of result and possibly knowing its class structure. It is fine for people who have the code memorized, but imagine a new contributor wanting to decode this and not even knowing where to start.

Say, `p.will_eat( it ).value` - it would clearly point to some class, as opposed to pointer or ID. Less terse, but clearer. Sometimes less isn't more.

Reply:
Me too have doubts about this decision. .value looks better.

Reply:
Agreed.